shap/explainers/_tree.py

```diff
@@ -1671,6 +1661,15 @@ def _recalculate_value(tree, i , level):
             self.features = np.where(self.features >= 0, tree_features[self.features], self.features)
 
 
+def get_dmatrix_properties(model: xgboost.sklearn.XGBModel):
```

Comment:
shap does not use any type hints yet, so not sure what the maintainers will say about this. But if you use them could you also please add the correct return type `Dict[str, Union[int, float, str, ...]]` (not sure if I missed something hence `...`. And import Dict and Union from typing? Since we need to support python3.7 and onwards